### PR TITLE
Modularize project

### PR DESCRIPTION
**Pull request by fxprunayre**
Reorganize project to multimodules with 

image

Centralize config for apps and tests

image

Experiment moving code from GN4 to GN5 - eg. Batch edit API to evaluate how to organize modules and separate legacy and new code.

### CODE REVIEW

**Comment by jodygarnett** [comment]
@josegar74 and @fxprunayre this repo is about to be moved. 

I am really interested in this PR :)

**Comment by fxprunayre** [comment]
> @josegar74 and @fxprunayre this repo is about to be moved.

Before moving, we should merge this PR.


> I am really interested in this PR :)

We need a maven expert here @jodygarnett . For Jose and me, build is all fine locally but not in github action. Can you give a try ? The issue is related to

```
Error:  Failed to execute goal on project gn-indexing: Could not resolve dependencies for project org.geonetwork:gn-indexing:jar:5.0.0-SNAPSHOT: The following artifacts could not be resolved: org.geonetwork:gn-testing:jar:tests:5.0.0-SNAPSHOT (absent): Could not find artifact org.geonetwork:gn-testing:jar:tests:5.0.0-SNAPSHOT -> [Help 1]
```

We did not figure out why last week.

**Comment by jodygarnett** [comment]
@fxprunayre [data-upload](https://github.com/GeoCat/geonetwork/tree/data-upload) branch will move along with this repository.

https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository 

> When you transfer a repository, its issues, pull requests, wiki, stars, and watchers are also transferred. If the transferred repository contains webhooks, services, secrets, or deploy keys, they will remain associated after the transfer is complete.

So we can move to the geonetwork organization today, and then I will work on fixing this data-upload branch with you next?

**Comment by jodygarnett** [comment]
> We did not figure out why last week.

I cannot reproduce this problem? I do have test failures, but building without tests works.

```
mvn clean install -DskipTests
```

The test failures are using:
```
mvn clean install -fae
```

File: pom.xml [hunk 1]
```diff
@@ -36,6 +37,12 @@
       <url>https://www.gnu.org/licenses/old-licenses/gpl-2.0.txt</url>
     </license>
   </licenses>
+
+  <modules>
+    <module>src/shared</module>
+    <module>src/modules</module>
+    <module>src/apps</module>
```
Comment (jodygarnett) [inline pom.xml:44]: Not sure what value the `src` is having here?  Is it a convention for spring-boot multi-module apps?

- Q: What is "shared"? Is it common code shared between several plugins?
- Q: What is "modules"? Everything in maven is a module; perhaps it is something else ...
-Q: To confirm plural "apps" and "modules"?


Recommend `shared`, `module`, `app`.

Comment (fxprunayre) [reply pom.xml:44]: No specific convention I think. @josegar74 ?

https://github.com/geonetwork/geonetwork-microservices/ was using another layout.

> Q: What is "shared"? Is it common code shared between several plugins?

Yes

> Q: What is "modules"? Everything in maven is a module; perhaps it is something else ...

We started naming it "extension" but we were not convinced :)
Modules will contain each components/features in the application ... 

> Q: To confirm plural "apps" and "modules"?

No strong opinion.

**Comment by jodygarnett** [comment]
nice I think I fixed it, the quick build has now passed.

**Comment by jodygarnett** [comment]
the default and full builds are getting stuck:
```
Error:  Errors: 
Error:    MetadataBuilderTest.dataAnalysisFromShapefileFromEmptyRecord:71 » DataAnalyzer Unable to find image 'ghcr.io/osgeo/gdal:alpine-normal-latest' locally
...
Error:  Tests run: 14, Failures: 0, Errors: 1, Skipped: 0
Error:  Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.2.5:test (default-test) on project gn-data-analysis: 
```

However I cannot reproduce this issue locally.

As it is referencing a docker environment I expect that MetadataBuilderTest may be some kind of integration test.

**Comment by jodygarnett** [comment]
@fxprunayre some CodeQL feedback, but we are good to go?

**Comment by jodygarnett** [comment]
@josegar74 this is good to go

**Pull request merged**